src/AWS.Deploy.ServerMode.Client/DeploymentCommunicationClient.cs

```diff
@@ -10,7 +10,20 @@
 
 namespace AWS.Deploy.ServerMode.Client
 {
-    public class DeploymentCommunicationClient : IDisposable
+    public interface IDeploymentCommunicationClient : IDisposable
+    {
+        Action<string>? ReceiveLogDebugLine { get; set; }
+
+        Action<string>? ReceiveLogErrorMessageLine { get; set; }
+
+        Action<string>? ReceiveLogMessageLineAction { get; set; }
+
+        Action<string>? ReceiveLogAllLogAction { get; set; }
+
+        Task JoinSession(string sessionId);
+    }
+
+    public class DeploymentCommunicationClient : IDeploymentCommunicationClient, IDisposable
```

Comment:
Why do we need to specify the IDisposable interface on both IDeploymentCommunicationClient and DeploymentCommunicationClient?

Reply:
Removed IDisposable from DeploymentCommunicationClient.